tests/test_series.py

```diff
@@ -1105,8 +1147,8 @@ def test_types_getitem() -> None:
     s = pd.Series({"key": [0, 1, 2, 3]})
     key: list[int] = s["key"]
     s2 = pd.Series([0, 1, 2, 3])
-    value: int = s2[0]
-    s3: pd.Series = s[:2]
+    check(assert_type(s2[0], int), np.integer)
+    check(assert_type(s[:2], pd.Series), pd.Series)
```

Comment:
We should add tests here for `s[0]`  (type check as `Any`, but is a list that is returned) and `s2[:2]`  (type check as `Series[int]`, check the same)

Reply:
Done, good point!

Reply:
Actually accessing s with `0` will cause an warning in python 3.11 and higher:
```
FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
```
I didi add the tests for `s2[:2]`